fetch.js

```diff
@@ -1,23 +1,47 @@
-(function(self) {
+(function (root, factory) {
   'use strict';
 
-  if (self.fetch) {
-    return
+  if (typeof define === 'function' && define.amd) {
+    // AMD. Register as an anonymous module.
+    define([], factory)
+  } else if (typeof module === 'object' && module.exports) {
+    // Node. Does not work with strict CommonJS, but
+    // only CommonJS-like environments that support module.exports,
+    // like Node.
+    module.exports = factory()
+  } else {
+    // Browser globals (root is window)
+    root.fetchPolyfill = factory()
+    if (!root.fetch) {
+      for (var attr in root.fetchPolyfill) {
+        root[attr] = root.fetchPolyfill[attr]
+      }
+    }
+  }
+}(this, function () {
+  'use strict';
+
+  var main = {};
+
+  try {
+    main = window
+  } catch(e) {
+    main = global
   }
 
   var support = {
-    searchParams: 'URLSearchParams' in self,
```

Comment:
This unnecessarily increases the size of the diff. Try naming the global object `self` instead of `main`. Also, please pass the global object by argument into the factory function instead of determining it inside the factory.